### DISCUSSION TITLE
Power consumption

### DISCUSSION THREAD

**Discussion by Tollgren** [onlaj/Piano-LED-Visualizer#172] [General]
The theoretical power consumption on a single ws2812b is 50-53mA at full brightness and white. 
I have a 61key piano i have (122 leds) and if i try Led animation breathing with the only power source USB-power i get a power consumption at 1.8A at max (measured it with a USB power meter). Theoretically it should be 6.1A + Power for the Pi.
The most realistic test a to be Scanner and it draws only 0,5A. 
The Raspberry Pi Zero has no fuse and can deliver all you put into the USB and transfer it to pin4 (5V) on the Pi. Well, at one point the Pi is going to start smoking if you send too much current through it. But normal Raspberry IP 3-4 can handle 2.5-5A. So a normal Android mobile charger 2.4A is perfect for this build. The big power brick suggested is not so fun...
Anyone else tested there build only with USB-power?

And if one is going for allot of white light (3x power consumption) one should consider SK6812 RGBW (instead of WS2812b RGB). SK6812 has 1 diode for white instead of 3. The colours are also a bit more pastel on a RGBW.  
![bild](https://user-images.githubusercontent.com/24440614/102475295-fc0e6180-4059-11eb-9410-a987de74a8f7.png)

**Comment by onlaj** [comment]
I just connected power meter between wall socket and led strip's power supply, set brightness to 100% and lighted all 172 leds with white color. The power usage went to 33W. Assuming power supply's 90% efficiency it would seem to be right because mine is 30W (5V 6A). It would go higher if I had better power supply, theoretically up to 50W.
I mentioned in the readme file that if you are not going to light more than 10 leds at the same time you should be fine powering it with RPi only. [Here](https://tutorials-raspberrypi.com/connect-control-raspberry-pi-ws2812-rgb-led-strips/) the autor wrote:
>If your strip has less than 20-30 LEDs, external power is not required.

Maybe I will add some software limit to prevent using more than X amps in case you don't have psu.

**Reply by Tollgren** [reply]
I saw now when I looked in Led strip settings on my Pi that Brightness was set to 50% 3.66A. But I do not know if it affects LED Animation and Breathing, and it only used 1.8A.
Maybe you should add an instruction on a capacitor of 5V 1000uF to reduce the effects at power consumption peaks?
It's normal in led projects.
Support for SK6812 can also reduce power consumption in addition to being a very nice led ;)
